### DISCUSSION TITLE
Smart Meter Interface: Connect Hichi USB Head via ESP32-S3 USB-OTG

### DISCUSSION THREAD

**Discussion by r-hof** [arendst/Tasmota#18365] [Q&A] [answered]
Hi Forum, the Tasmota "Smart Meter Interface" extension might be the right tool for reading out my power meter. However, I cannot wire the Hichi IR reading device directly to my ESP32-S3 DevKit's GPIO Pins as described in the extension's documentation, instead I would like to connect the Hichi USB variant to the ESP32-S3 USB port.

Has anybody already tried this and has a working solution? I am an absolute beginner in the ESP32/Microcontroller world and my first encounters with Espressif's USB example code have been very frustrating. Any help appreciated.

**Comment by sfromis** [comment, marked as answer]
Tasmota does not have support for using USB for more than serial terminal (including USB CDC).
